MapboxNavigation/NavigationMapView.swift

```diff
@@ -54,55 +51,252 @@ open class NavigationMapView: MGLMapView {
     dynamic public var trafficSevereColor: UIColor = .trafficSevere
     dynamic public var routeCasingColor: UIColor = .defaultRouteCasing
     
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        makeGestureRecognizersRespectCourseTracking()
+        makeGestureRecognizersUpdateCourseView()
+        
+        UIDevice.current.addObserver(self, forKeyPath: "batteryState", options: [.initial, .new], context: nil)
+    }
+    
+    public required init?(coder decoder: NSCoder) {
+        super.init(coder: decoder)
+        
+        makeGestureRecognizersRespectCourseTracking()
+        makeGestureRecognizersUpdateCourseView()
+        
+        UIDevice.current.addObserver(self, forKeyPath: "batteryState", options: [.initial, .new], context: nil)
+    }
+    
+    /** Modifies the gesture recognizers to also disable course tracking. */
+    func makeGestureRecognizersRespectCourseTracking() {
+        for gestureRecognizer in gestureRecognizers ?? []
+            where gestureRecognizer is UIPanGestureRecognizer || gestureRecognizer is UIRotationGestureRecognizer {
+                gestureRecognizer.addTarget(self, action: #selector(disableUserCourseTracking))
+        }
+    }
+    
+    func makeGestureRecognizersResetInactivityTimer() {
+        for gestureRecognizer in gestureRecognizers ?? [] {
+            gestureRecognizer.addTarget(self, action: #selector(resetInactivityTimer(_:)))
+        }
+    }
+    
+    func makeGestureRecognizersUpdateCourseView() {
+        for gestureRecognizer in gestureRecognizers ?? [] {
+            gestureRecognizer.addTarget(self, action: #selector(updateCourseView(_:)))
+        }
+    }
+    
+    deinit {
+        UIDevice.current.removeObserver(self, forKeyPath: "batteryState")
+    }
+    
+    open override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
+        if keyPath == "batteryState" {
+            let batteryState = UIDevice.current.batteryState
+            isPluggedIn = batteryState == .charging || batteryState == .full
+        }
+    }
+    
+    func updateCourseView(_ sender: UIGestureRecognizer) {
+        
+        // Disable course tracking when any kind of user interaction occurs
+        if tracksUserCourse {
```

Comment:
Why disable course tracking here ?
Isn't it the purpose of the selector: `disableUserCourseTracking` which is only triggered by rotation and pan gesture recognizers ?